framework/include/userobjects/TimedElementSubdomainModifier.h

```diff
@@ -0,0 +1,51 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+#include "ElementSubdomainModifier.h"
+
+/**
+ * Modifies element subdomains only at a given list of times
+ */
+class TimedElementSubdomainModifier : public ElementSubdomainModifier
+{
+public:
+  static InputParameters validParams();
+
+  TimedElementSubdomainModifier(const InputParameters & parameters);
+
+  virtual void initialize() override;
+
+protected:
+  /**
+   * Requests a vector of all points in time from the inheriting class
```

Comment:
```suggestion
   * Requests a vector of all times from the inheriting class
```

“Points” is potentially misleading.